drivers/i2c/target/eeprom_target.c

```diff
@@ -129,7 +130,7 @@ static int eeprom_target_write_received(struct i2c_target_config *config,
 
 		data->buffer_idx = val | (data->buffer_idx << 8);
 		data->idx_write_cnt++;
-	} else {
+	} else if (!data->read_only) {
```

Comment:
This is insufficient. The I2C write transaction would need to be NAK'ed as well (see the comment just above the code, you've changed).

Reply:
Thanks for pointing it out, of course you are right